dhis2/dhis2-core
# fix: only restrict set of checks when executing them [DHIS2-16256]

### Summary
When executing data integrity checks the default (no set of checks was explicitly specified) is to only run checks that are not marked as "slow". This filter wrongly was also used when expanding checks to read the results from the cache. This meant that slow checks would not be included unless explicitly named.

### Automatic Testing
Modified existing tests with additional checks that verify that reading check descriptions and check summaries after execution does include the slow checks which means they are not filtered.

### Manual Testing
* run checks POST `/api/dataIntegrity/summary` (no `checks` parameter to run all defaults)
* run a specific slow check POST `/api/dataIntegrity/summary?checks=data_elements_aggregate_abandoned`
* wait some
* fetch results GET `/api/dataIntegrity/summary` should include the slow check `data_elements_aggregate_abandoned`
